controlplane/kubeadm/internal/workload_cluster_coredns_test.go

```diff
@@ -280,6 +286,131 @@ kind: ClusterConfiguration
 			objs:          []runtime.Object{depl, cm, kubeadmCM},
 			expectErr:     false,
 			expectUpdates: true,
+			expectImage:   "k8s.gcr.io/some-repo/coredns:1.7.2",
+		},
+		{
+			name: "updates everything successfully to v1.8.0 with a custom repo should not change the image name",
+			kcp: &controlplanev1.KubeadmControlPlane{
+				Spec: controlplanev1.KubeadmControlPlaneSpec{
+					KubeadmConfigSpec: cabpkv1.KubeadmConfigSpec{
+						ClusterConfiguration: &kubeadmv1.ClusterConfiguration{
+							DNS: kubeadmv1.DNS{
+								Type: kubeadmv1.CoreDNS,
+								ImageMeta: kubeadmv1.ImageMeta{
+									// provide an newer image to update to
+									ImageRepository: "k8s.gcr.io/some-repo",
```

Comment:
Wondering if this test should probably use an image repo that is not k8s.gcr.io in this test

Reply:
Would have been a bit better